src/theory/quantifiers/term_tuple_enumerator.h

```diff
@@ -0,0 +1,57 @@
+/*********************                                                        */
+/*! \file  term_tuple_enumerator.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Mikolas Janota
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Implementation of an enumeration of tuples of terms for the purpose
+ *of quantifier instantiation.
+ **/
+#ifndef TERM_TUPLE_ENUMERATOR_H_7640
+#define TERM_TUPLE_ENUMERATOR_H_7640
+#include <vector>
+
+#include "smt/smt_statistics_registry.h"
+#include "theory/quantifiers/relevant_domain.h"
+#include "theory/quantifiers_engine.h"
```

Comment:
Can you forward declare QuantifiersEngine and then include it in the cpp?  We are trying to eliminate our use of this include.